extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -9,69 +9,195 @@
+    private final ScheduledExecutorService scheduledExecutorService;
+    @NotNull
     private final ObjectMapper objectMapper;
+    @NotNull
+    private final Monitor monitor;
 
-    HashicorpVaultClient(@NotNull HashicorpVaultClientConfig hashicorpVaultConfig, @NotNull EdcHttpClient httpClient,
-                         @NotNull ObjectMapper objectMapper) {
+    HashicorpVaultClient(@NotNull HashicorpVaultConfig hashicorpVaultConfig,
+                         @NotNull EdcHttpClient httpClient,
+                         @NotNull ObjectMapper objectMapper,
+                         @NotNull ScheduledExecutorService scheduledExecutorService,
+                         @NotNull Monitor monitor) {
         this.hashicorpVaultConfig = hashicorpVaultConfig;
         this.httpClient = httpClient;
         this.objectMapper = objectMapper;
+        this.scheduledExecutorService = scheduledExecutorService;
+        this.monitor = monitor;
+    }
+
+    Result<HealthCheckResponse> doHealthCheck() {
+        var requestUri = getHealthCheckUrl();
+        var request = httpGet(requestUri);
+
+        try (var response = httpClient.execute(request)) {
+            var code = response.code();
+            var responseBody = Objects.requireNonNull(response.body()).string();
+            var responsePayload = objectMapper.readValue(responseBody, HealthCheckResponsePayload.class);
+            var healthCheckResponse = HealthCheckResponse.Builder
+                    .newInstance()
+                    .code(code)
+                    .payload(responsePayload)
+                    .build();
+            return Result.success(healthCheckResponse);
+        } catch (IOException e) {
+            return Result.failure(e.getMessage());
+        }
+    }
+
+    void inspectToken(TokenLookUpResponsePayloadToken token) {
+        if (token.isRootToken()) {
+            monitor.warning("Root token detected. Don't use root tokens in production environment!");
+        } else {
+            if (token.isRenewable()) {
+                monitor.info("Token is renewable");
+            } else {
+                monitor.warning("Token is not renewable");
+            }
+
+            if (token.isPeriodicToken()) {
+                monitor.info("Token has a renewal period of %s".formatted(token.getPeriod()));
+                monitor.warning("Configured time-to-live (ttl) will not be honored for periodic tokens");
+            } else {
+                monitor.warning("Non-periodic Token will expire at some point. Check 'max_ttl' property inside Vault server configuration file for more information");
+            }
+
+            if (token.hasExplicitMaxTimeToLive()) {
+                var explicitMaxTtl = token.getExplicitMaxTimeToLive();
+                var formattedTime = TIME_FORMAT.formatted(explicitMaxTtl / 3600, (explicitMaxTtl % 3600) / 60, explicitMaxTtl % 60);
+                monitor.warning("Token will ultimately expire in %s".formatted(formattedTime));
```

Comment:
Again, the same issues as above. Why is this a warning? What will happen? What token is it?